in-toto-spec.md

```diff
@@ -529,106 +539,73 @@ To provide descriptive examples, we will adopt "canonical JSON," as described
-ROLE.
+#### 4.2.1 Key formats
 
 The current reference implementation of in-toto defines three signing methods,
 although in-toto is not restricted to any particular key signing method, key
 type, or cryptographic library:
 
-* "RSASSA-PSS" : RSA Probabilistic signature scheme with [appendix](http://tools.ietf.org/html/rfc3447#page-29).
+* "RSASSA-PSS": RSA Probabilistic signature scheme with [appendix](http://tools.ietf.org/html/rfc3447#page-29).
 The underlying hash function is SHA256.
-* "ed25519" : Elliptic curve digital signature algorithm based on
+* "ed25519": Elliptic curve digital signature algorithm based on
     [Twisted Edwards curves](http://ed25519.cr.yp.to/).
-* "ecdsa" : [Elliptic curve digital signature algorithm](https://tools.ietf.org/html/rfc6979)
+* "ecdsa": [Elliptic curve digital signature algorithm](https://tools.ietf.org/html/rfc6979)
 
 All keys have the format:
 
 ```json
-  { "keytype" : "<KEYTYPE>",
-    "scheme" : "<SCHEME>",
-    "keyval" : "<KEYVAL>" }
-```
-
-KEYTYPE is a string denoting a public key signature system, such as RSA or
-ECDSA. SCHEME is a string denoting a corresponding signature scheme. For
-example: "rsassa-pss-sha256" and "ecdsa-sha2-nistp256". KEYVAL is a dictionary
-containing the public portion of the key.
-
-We define three key types at present: "rsa", "ed25519", and "ecdsa".
-
-The 'rsa' format is:
-
-```json
-  { "keytype" : "rsa",
-    "scheme" : "rsassa-pss-sha256",
-    "keyval" : { "public" : "<PUBLIC>",
-                 "private" : "<PRIVATE>" }
-  }
+{
+  "keytype" : "<KEYTYPE>",
+  "scheme" : "<SCHEME>",
+  "keyval" : "<KEYVAL>"
+}
 ```
 
-where PUBLIC and PRIVATE are in PEM format and are strings.  All RSA keys must
-be at least 2048 bits.
-
-The elliptic-curve variants ("ed25519" and "ecdsa") format are:
-
-```json
-  { "keytype" : "ed25519",
-    "scheme" : "ed25519",
-    "keyval" : { "public" : "<PUBLIC>",
-                 "private" : "<PRIVATE>" }
-  }
-```
+KEYTYPE is a string denoting a public key signature system. SCHEME is a string
+denoting a corresponding signature scheme. KEYVAL is a dictionary containing the
+public portion of the key. The following table summarizes the expected entries
+for each signing algorithm supported by the reference implementation.
 
-```json
-  { "keytype" : "ecdsa",
-    "scheme" : "ecdsa-sha2-nistp256",
-    "keyval" : { "public" : "<PUBLIC>",
-                 "private" : "<PRIVATE>" }
-  }
-```
+| Method | KEYTYPE | SCHEME| KEYVAL | Notes |
+|--------|---------|-------|--------|-------|
+| RSASSA-PSS | rsa | rsassa-pss-sha256 | PEM formatted string | RSA keys must be at least 2048 bits |
+| ED25519 | ed25519 | ed25519 | 256 bit strings | |
+| ECDSA | ecdsa | ecdsa-sha2-nistp256 | 256 bit strings | |
 
-where PUBLIC and PRIVATE are both 32-byte (256-bit) strings.
+Note: This key format is for representation of keys in in-toto metadata only.
+Implementations may serialize keys to disk using any format they prefer.
```

Comment:
Minor nit: Emphasizing that this is the key metadata format is a good idea. But I don't think we need to suggest, that in-toto users serialize keys to disk. With the new securesystemslib signer API they could be using HSMs or cloud signing providers (see https://github.com/in-toto/in-toto/issues/533, https://github.com/in-toto/in-toto/issues/532).

Reply:
removed the second line altogether.